### DISCUSSION TITLE
人称问题在日翻中情景下确实无法解决。

### DISCUSSION THREAD

**Discussion by daiaji** [SakuraLLM/SakuraLLM#88] [General]
是的，对于大语言模型（LLM）来说，处理日语中的人称代词和宾语省略确实是一个挑战。以下是一些具体的原因和应对策略：

### 挑战原因：

1. **上下文依赖性强：**
   * 日语中人称代词和宾语的省略依赖于上下文，模型需要理解整个上下文才能准确填补省略的部分。
   
2. **性别判断困难：**
   * 由于省略了人称代词，模型需要根据其他线索（如角色名称、语境、对话内容）来判断性别，这增加了复杂性。
   
3. **文化和语言差异：**
   * 日语和中文在表达方式和文化背景上有很大差异，这可能导致模型在翻译时出现误解或错误。

### 应对策略：

1. **增强上下文理解：**
   * 通过训练模型处理更长的上下文，增强其对上下文的理解能力，使其能够更好地判断省略的部分。

2. **增加训练数据：**
   * 提供更多包含人称代词和宾语省略的日语-中文翻译对，帮助模型学习如何在不同语境下填补省略的部分。

3. **引入性别判断机制：**
   * 通过在训练数据中包含明确的性别信息，或者利用其他语言特征（如角色名称和对话内容）来帮助模型进行性别判断。

4. **多轮对话模拟：**
   * 通过模拟多轮对话，训练模型在对话过程中逐步推断出省略的部分，增强其处理连续对话的能力。

### 示例总结：

**原文：**
1. **“ヴァラ、という人物ですが……”**
   * 翻译: "叫Vala，这个人……"
2. **“ヴァラさん！　ヴァラさんのことなら知ってる！”**
   * 翻译: "Vala！我知道Vala！"
3. **“なるほど、マスターはヴァラに発情しているというわけですか。”**
   * 翻译: "原来如此，主人对Vala动了情。"

**翻译策略：**
1. **理解上下文：** 结合上下文，推断省略的部分。
2. **判断性别：** 利用对话内容和角色名称等线索判断性别。
3. **避免冗长：** 尽量省略不必要的人称代词，保持翻译简洁。

### 总结：

大语言模型在处理日语中的人称代词和宾语省略时确实面临挑战，但通过增强上下文理解、增加训练数据和引入性别判断机制等策略，可以提高其翻译准确性。
___
看起来基本上只能提升模型推理能力和上下文长度来补救。
而且有时候可能推理也没用。
只能打完游戏来校对。

上下文长度，gemini这种以量大著称的LLM似乎也只能70行。
再多，输出的数量就会和原文不匹配。
这个至少现在应该是做不到。

**Comment by sakura-umi** [comment]
感谢您的分析和建议！虽然目前难度确实较大，但我们仍然会尝试不同的办法试图缓解此类问题，或者至少降低使用模型的汉化者校对人称的时间成本。

**Reply by daiaji** [reply]
人名统一问题同样需要长上下文和推理能力增强，现阶段应该也搞不定。
指令理解、执行是要换个更好参数更大的底模吧，qwen现在还是打不过llama3。
72B级别llama3比qwen指令执行还是强很多。

**Reply by sakura-umi** [reply]
只是人名统一的话很好解决，目前正在实验中的版本（0.10pre2）已经可以达到很高的成功率，之后会考虑发布。

而理解人名的注释（如性别，职业等）这个方面是难点，目前只能做一些实验寻找有用的办法，已经有一个有一定效果的方法，但人力成本过高目前无法实现。但这个问题相对地影响没有人名统一那么大，所以并不着急。

实际上单独修正他/她的性别人称也较容易解决，在之前的0.10pre0版本里已经可以有一定的成功率。

**Reply by daiaji** [reply]
> 只是人名统一的话很好解决，目前正在实验中的版本（0.10pre2）已经可以达到很高的成功率，之后会考虑发布。

暂存吗？
>实际上单独修正他/她的性别人称也较容易解决，在之前的0.10pre0版本里已经可以有一定的成功率。

单独修正没啥用吧，有时一段翻译文本里很多情况就是省略人称代词，只有一段文本的话，只能硬推理。

还是说推理出人物的性别也暂存？但推理一开始就是错的呢？

**Reply by sakura-umi** [reply]
不懂你说的暂存是什么意思

**Comment by dantecsm** [comment]
llm 跑完后第三人称代词性别不分的问题，有啥合适的问题定位和纠正方法吗？
不是从模型角度解决，而是在模型输出之后，我们现在拿到的结果的基础上后处理呢？
